python/hail/table.py

```diff
@@ -2138,6 +2138,52 @@ def row(self):
         # FIXME: 'row' and 'globals' symbol in the Table parser, like VSM
         raise NotImplementedError()
 
+    @staticmethod
+    @handle_py4j
+    @typecheck(df=DataFrame,
+               key=oneof(strlike, listof(strlike)))
+    def from_spark(df, key=[]):
+        """Convert PySpark SQL DataFrame to a table.
+
+        Examples
+        --------
+
+        >>> t = Table.from_spark(df) # doctest: +SKIP
+
+        Notes
+        -----
+
+        Spark SQL data types are converted to Hail types as follows:
+
+        .. code-block:: text
+
+          BooleanType => Boolean
```

Comment:
can you link to our types here instead of a code block?

also, we don't expose binary in Python. that could be a problem.

Reply:
Will the links work in a code block?

I don't think we use Binary anymore, but it is pretty useful and I'm sure there are people using 0.1 using it.  I think we should support it.  Now that we're in python3, it should be treated as a bytes-like object.  I'll leave the mapping in for now and add an issue to add back TBinary to python.